src/main/java/org/jfrog/hudson/util/MavenVersionHelper.java

```diff
@@ -112,10 +114,13 @@ private static boolean isAtLeast(MavenModuleSetBuild build, String mavenHome, St
     private static String getMavenVersion(MavenModuleSetBuild build, EnvVars vars,
             BuildListener listener) throws IOException, InterruptedException {
         final Maven.MavenInstallation installation = getMavenInstallation(build, vars, listener);
-        return build.getWorkspace().act(new FilePath.FileCallable<String>() {
-            public String invoke(File f, VirtualChannel channel) throws IOException, InterruptedException {
+        final String home = installation.getHome();
+
+        return build.getWorkspace().act(new MasterToSlaveCallable<String, IOException>() {
+            @Override
+            public String call() throws IOException {
                 try {
-                    return MavenEmbedderUtils.getMavenVersion(new File(installation.getHome())).getVersion();
```

Comment:
How could this have worked before?? `ToolInstallation`s are not supposed to be serialized AFAIK.

Reply:
I think that Javac was smart enough to calculate `installation.getHome()` on the master side and send only results